crates/holochain/src/core/workflow/call_zome_workflow.rs

```diff
@@ -75,38 +75,48 @@ where
     if should_write {
         let is_empty = workspace.source_chain().is_empty()?;
         let countersigning_op = workspace.source_chain().countersigning_op()?;
-        let flushed_actions: Vec<SignedActionHashed> = HostFnWorkspace::from(workspace.clone())
+        match HostFnWorkspace::from(workspace.clone())
             .flush(&network)
-            .await?;
-        if !is_empty {
-            match countersigning_op {
-                Some(op) => {
-                    if let Err(error_response) =
-                        super::countersigning_workflow::countersigning_publish(&network, op).await
-                    {
-                        return Ok(Ok(error_response));
+            .await
+        {
+            Ok(flushed_actions) => {
+                // Q: what is the purpose of checking for an empty chain? When would this ever happen? The chain should
```

Comment:
I agree. Are you hesitating to remove this check?

Reply:
I looked into it more, and we have a slight inconsistency in the `SourceChain` implementation now, I added a FIXME about it elsewhere. It feels important enough to be aware of but maybe not important enough to fix right now.